tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,373 @@
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing customVol",
+			Namespace: "customvol",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/customVol/db.yaml",
+				ProjectPath + "docs/examples/customVol/web.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing envFrom",
+			Namespace: "envfrom",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/envFrom/db.yaml",
+				ProjectPath + "docs/examples/envFrom/web.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing extraResources",
+			Namespace: "extraresources",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/extraResources/app.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "web", Port: 80},
+			},
+		},
+		{
+			TestName:  "Testing health",
+			Namespace: "health",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/health/db.yaml",
+				ProjectPath + "docs/examples/health/web.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing healthChecks",
+			Namespace: "healthchecks",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/healthchecks/db.yaml",
+				ProjectPath + "docs/examples/healthchecks/web.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing secret",
+			Namespace: "secrets",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/secrets/db.yaml",
+				ProjectPath + "docs/examples/secrets/web.yaml",
+			},
+			PodStarted: []string{"web"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Testing single file",
+			Namespace: "singlefile",
+			InputFiles: []string{
+				ProjectPath + "docs/examples/single_file/wordpress.yml",
+			},
+			PodStarted: []string{"wordpress"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 8080},
+			},
+		},
+		{
+			TestName:  "Normal Wordpress test",
+			Namespace: "wordpress",
+			InputFiles: []string{
+				ProjectPath + "examples/wordpress/wordpress.yaml",
+				ProjectPath + "examples/wordpress/mariadb.yaml",
+			},
+			PodStarted: []string{"wordpress"},
+			NodePortServices: []ServicePort{
+				{Name: "wordpress", Port: 80},
+			},
+		},
+	}
+
+	_, err = clientset.CoreV1().Pods("").List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Kubernetes cluster is not running or not accessible: %v", err)
+	}
+
+	for _, test := range tests {
+		test := test // capture range variable
+		t.Run(test.TestName, func(t *testing.T) {
+			t.Parallel()
```

Comment:
See: https://golang.org/pkg/testing/ with t.Parallel() regardless of what you put as GOMAXPROCS it still runs everything at the same time. Deploying *every* test to Kubernetes.

Perhaps removing this would help...

Even with me setting the GOMAXPROCS to something lower, I still have every test being deploying at the same time to Kubernetes, which isn't helping.. There should be a limiter to how many tests are ran at a time.

Reply:
@cdrage I think I know what is confusing you, follow the following steps.

I get this when I run tests:

```bash
$ make test-e2e PARALLEL=2
go test -parallel=2 -v github.com/kedgeproject/kedge/tests/e2e
=== RUN   Test_Integration
=== RUN   Test_Integration/Testing_configMap
=== RUN   Test_Integration/Testing_customVol
=== RUN   Test_Integration/Testing_envFrom
=== RUN   Test_Integration/Testing_extraResources
=== RUN   Test_Integration/Testing_health
=== RUN   Test_Integration/Testing_healthChecks
=== RUN   Test_Integration/Testing_secret
=== RUN   Test_Integration/Testing_single_file
=== RUN   Test_Integration/Normal_Wordpress_test
...
```

Above output shows that all of them are running, but

In another window, while above is happening I could see:

```bash
$ kubectl get ns -w
NAME          STATUS    AGE
default       Active    14d
kube-public   Active    14d
kube-system   Active    14d
configmap   Active    0s
singlefile   Active    0s
configmap   Terminating   16s
wordpress   Active    0s
configmap   Terminating   38s
configmap   Terminating   38s
wordpress   Terminating   33s
health    Active    0s
...
```

In above output you can see, `configmap` and `singlefile` were created; then `configmap` was done and `wordpress` was started and so on. So at anytime only two tests at maximum are running.

Reply:
Also from docs and help:

```bash
$ go test -h
test [build/test flags] [packages] [build/test flags & test binary flags]

In addition to the build flags, the flags handled by 'go test' itself are:

        -args

...


        -parallel n
            Allow parallel execution of test functions that call t.Parallel.
            The value of this flag is the maximum number of tests to run
            simultaneously; by default, it is set to the value of GOMAXPROCS.
            Note that -parallel only applies within a single test binary.
            The 'go test' command may run tests for different packages
            in parallel as well, according to the setting of the -p flag
            (see 'go help build').

...
```